packages/matter.js/src/node/ServerNode.ts

```diff
@@ -57,7 +62,15 @@ export class ServerNode<T extends ServerNode.RootEndpoint = ServerNode.RootEndpo
     constructor(config: Partial<Node.Configuration<T>>);
 
     constructor(definition?: T | Node.Configuration<T>, options?: Node.Options<T>) {
-        super(Node.nodeConfigFor(ServerNode.RootEndpoint as T, definition, options));
+        const config = Node.nodeConfigFor(ServerNode.RootEndpoint as T, definition, options);
+
+        // Enable reachableChanged event on the BasicInformation cluster of Root endpoint if reachable attribute is defined
+        if (config.type.deviceType === RootEndpoint.deviceType && config.basicInformation?.reachable !== undefined) {
+            config.type = config.type.with(
+                BasicInformationServer.enable({ events: { reachableChanged: true } }),
```

Comment:
The DeviceType in the configuration may specify an altered variant of BasicInformationServer. So this should probably be something along the lines of `config.type.behaviors.basicInformation.enable(...)`

Reply:
Irrelevant now